lazyflow/stype.py

```diff
@@ -100,13 +101,21 @@ def allocateDestination( self, roi ):
         shape = roi.stop - roi.start if roi else self.slot.meta.shape
         storage = numpy.ndarray(shape, dtype=self.slot.meta.dtype)
 
+        # if self.slot.meta.axistags is True:
+        #     storage = vigra.taggedView(storage, self.slot.meta.axistags)
+        #     #storage = vigra.VigraArray(storage, storage.dtype, axistags = copy.copy(self.slot.meta.axistags))))
+        #     ##storage = storage.view(vigra.VigraArray)
+        #     ##storage.axistags = copy.copy(self.slot.meta.axistags)
         if self.slot.meta.has_mask:
             storage_mask = numpy.zeros(storage.shape, dtype=bool)
-            storage = numpy.ma.masked_array(storage, mask=storage_mask, shrink=False)
-        # if axistags is True:
-        #     storage = vigra.VigraArray(storage, storage.dtype, axistags = copy.copy(s))elf.axistags))
-        #     #storage = storage.view(vigra.VigraArray)
-        #     #storage.axistags = copy.copy(self.axistags)
+            storage_fill_value = None
```

Comment:
@stuarteberg I would be interested in your feedback here. I have chosen a standard different from NumPy for a few reasons. Of course, these are only the defaults, so the user (i.e. developer) can change them anyways.

1. NumPy picks values that don't make sense for certain types ( http://docs.scipy.org/doc/numpy/reference/maskedarray.baseclass.html#numpy.ma.MaskedArray.fill_value ). In particular, note the value for `int` is used for all integers (i.e. `uint8`), the value for `float` is out of range for `float16` and smaller.
2. It is nice to be able to have the same effect of having masked values even when they aren't there (viz. `nan`). This allows `numpy.isnan` to get back the mask or comparing to `0` in the integer case.